### DISCUSSION TITLE
Shorter disconnect timeout

### DISCUSSION THREAD

**Discussion by chakaz** [geckosio/geckos.io#98] [Help / Questions]
Hi all,
I'm trying Geckos for my in-development online game. So far it works spectacularly, so thanks a ton!
The one issue I'm seeing though is that it takes a **very** long time for Geckos (or more precisely the underlying webrtc implementation) to detect clients which disconnected. In my specific game this is a big deal, as in some cases players need to wait for all other players to reach some state.
Is there a way to set a shorter timeout / deadline for disconnection detection?
Thanks in advance!

**Comment by yandeu** [comment]
Have you tried version 2? Should be much faster.

**Reply by chakaz** [reply]
Haven't yet tried it, it's good to know that it should be faster.

Is there some estimation for when it will be officially released? Is it unstable / buggy at the moment?

**Reply by meisterpeeps** [reply]
@chakaz 
I have not implemented it yet for my game , but I intend to implement some sort of manual health-check for the connection.  I will have the server ping the client every second or so, and if the client doesn't respond within a certain time frame, assume they have disconnected.
